### PR TITLE
Add test case apache_ssl.pm

### PR DESCRIPTION
**Pull request by QingmingSu**
Test Apache server works with SSL enabled.

### CODE REVIEW

**Comment by QingmingSu** [comment]
Local test run at:
http://147.2.212.224/tests/330

File: tests/x11/apache_ssl.pm [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    # Start firefox and load the testing page
+    select_console 'x11';
+    ensure_unlocked_desktop;
+    x11_start_program("firefox https://localhost/hello.html");
+    assert_screen "firefox-ssl-untrusted", 60;
+    assert_and_click "firefox-ssl-advanced";
+    assert_and_click "firefox-ssl-addexception";
+    assert_screen "firefox-ssl-confirm-exception", 60;
+    send_key "alt-c";
+    assert_screen 'firefox-ssl-load-hellopage', 60;
```
Comment (okurz) [inline tests/x11/apache_ssl.pm]: I guess there should be something more lightweight than firefox to test a local ssl configuration in firefox. What about a curl call?

File: tests/x11/apache_ssl.pm [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+use base "x11test";
+use testapi;
+use utils;
+use strict;
+
+sub quit_firefox() {
+    send_key "alt-f4";
+    if (check_screen("firefox-save-and-quit", 10)) {
+        send_key "ret";
+    }
```
Comment (okurz) [inline tests/x11/apache_ssl.pm]: [DRY](https://en.wikipedia.org/wiki/Don't_repeat_yourself). This is a copy-paste from another module, but see comment below first before trying to change that.

File: tests/x11/apache_ssl.pm (hunk 1 above)
Comment (QingmingSu) [inline tests/x11/apache_ssl.pm]: Yeah, using curl command is more lightweight and easier to implement.
But I thought using firefox was more visible for end user to check the https process with untrusted certificate.

Comment (coolo) [inline tests/x11/apache_ssl.pm]: but firefox is also much more fragile. *much more*

**Comment by QingmingSu** [comment]
The test case codes have been updated to meet the change request.
Firefox part is removed, "curl -k " command is used now to verify the apache+ssl works.

New local test run can be found at:
http://147.2.212.224/tests/334#step/apache_ssl/1

**Comment by coolo** [comment]
:+1:

**Comment by coolo** [comment]
https://openqa.suse.de/tests/638644#step/apache_ssl/9 - timeout is too low it seems

**Comment by QingmingSu** [comment]
Wow, it is embarrassing. I will add timeout to 600s and test it on a heavy-loaded worker locally first.

**Pull request merged**